index.js

```diff
@@ -3,28 +3,51 @@ const process = require('process')
 const path = require('path')
 const spawn = require('child_process').spawnSync
 
-try {
-    // this job has nothing to do on non-Windows platforms
-    if (process.platform != 'win32') {
-        process.exit(0)
+/// Gets the inputs to this action, with default values filled in.
+function getInputs() {
+    // Default to the native processor as the host architecture
+    // vsdevcmd accepts both amd64 and x64
+    const hostArch = core.getInput('host_arch') || process.env['PROCESSOR_ARCHITECTURE'].toLowerCase() // amd64, x86 or arm64
+    const arch = core.getInput('arch') || hostArch
+    const toolsetVersion = core.getInput('toolset_version') || null
+
+    const components = core.getInput('components').split(';').filter(s => s.length != 0)
+    if (!toolsetVersion) {
+        // Include the latest target architecture compiler toolset by default
+        if (arch === 'arm64') {
+            components.push('Microsoft.VisualStudio.Component.VC.Tools.ARM64')
+        }
+        else if (arch == 'arm') {
+            components.push('Microsoft.VisualStudio.Component.VC.Tools.ARM')
+        }
+        else {
+            components.push('Microsoft.VisualStudio.Component.VC.Tools.x86.x64')
+        }
     }
 
-    const arch = core.getInput('arch') || 'amd64'
-    const hostArch = core.getInput('host_arch') || ''
-    const toolsetVersion = core.getInput('toolset_version') || ''
-    const winsdk = core.getInput('winsdk') || ''
-    const vswhere = core.getInput('vswhere') || 'vswhere.exe'
-    const components = core.getInput('components') || 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64'
-    const verbose = core.getInput('verbose') || false
+    return {
+        "host_arch": hostArch,
+        "arch": arch,
+        "toolset_version": toolsetVersion,
+        "winsdk": core.getInput('winsdk') || null,
+        "vswhere": core.getInput('vswhere') || null,
+        "components": components,
+        "verbose": String(core.getInput('verbose')) === "true"
```

Comment:
Does `github.core.getInput` automatically `toLowerCase()` the input?

Reply:
I doubt it. The reason I'm doing it this way is that I don't know what the result is if we call the action with `verbose: true` given that action inputs are stringly-typed. I consider it a bug if the input isn't "true" or "false" (or their boolean counterparts if that's possible)

Reply:
I think that `TRUE` should also be considered a proper input. This is stringly typed after all. Perhaps we should just do a `Boolean(core.getInput('verbose'))` instead?

Reply:
I tried that first
![image](https://github.com/user-attachments/assets/9c235440-f946-4093-8c63-462739462749)